src/ray/raylet/node_manager.cc

```diff
@@ -1098,24 +1113,30 @@ void NodeManager::HandleDisconnectedActor(const ActorID &actor_id, bool was_loca
       actor_registration.GetRemainingReconstructions() > 0 && !intentional_disconnect
           ? ActorTableData::RECONSTRUCTING
           : ActorTableData::DEAD;
-  if (was_local) {
-    // Clean up the dummy objects from this actor.
-    RAY_LOG(DEBUG) << "Removing dummy objects for actor: " << actor_id;
-    for (auto &dummy_object_pair : actor_entry->second.GetDummyObjects()) {
-      HandleObjectMissing(dummy_object_pair.first);
-    }
-  }
-  // Update the actor's state.
   ActorTableData new_actor_info = actor_entry->second.GetTableData();
   new_actor_info.set_state(new_state);
-  if (was_local) {
-    // If the actor was local, immediately update the state in actor registry.
-    // So if we receive any actor tasks before we receive GCS notification,
-    // these tasks can be correctly routed to the `MethodsWaitingForActorCreation`
-    // queue, instead of being assigned to the dead actor.
-    HandleActorStateTransition(actor_id, ActorRegistration(new_actor_info));
+
+  if (!was_local) {
+    auto actor_notification = std::make_shared<ActorTableData>(std::move(new_actor_info));
+    RAY_CHECK_OK(
+        gcs_client_->Actors().AsyncUpdate(actor_id, actor_notification, nullptr));
```

Comment:
If `was_local` is false, then this corresponds to a node failure, which should already have been handled by gcs server with this PR?

Reply:
> If `was_local` is false, then this corresponds to a node failure, which should already have been handled by gcs server with this PR?

The `HandleDisconnectedActor` function will never be called when gcs_service_enabled.

Reply:
Okay, could you add a comment in the header file for this function? It becomes a bit difficult to figure out which are used in gcs mode and which are not.